WordPress/gutenberg
# [RNMobile] Add integration test helpers

## What?


This PR incorporates a set of helper functions for integration tests.

**NOTE:** I'm planning to also move the helpers from [`block-draggable/test/helpers.native.js`](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/block-draggable/test/helpers.native.js) in another PR.

## Why?


This is part of an ongoing effort for making easier writing integration tests for the native version.

## How?


The helpers have been extracted mainly from the Gallery block test cases, as it incorporated several of them. A new folder named `integration-test-helpers` has been added to `test/native` path, in this folder we'll hold the helpers, ideally one file per helper.

In this PR we're incorporating the following ones, some of them were already being exposed from `test/native/helpers.js` file:
- `addBlock` - Adds a block via the block picker.
- `advanceAnimationByTime`, `advanceAnimationByFrame` - Advance Reanimated animations by time or frames.
- `dismissModal` - Dismisses a modal.
- `getBlock` - Gets a block from the root block list.
- `getEditorHtml` - Gets the current HTML output of the editor.
- `getInnerBlock` - Gets an inner block from another block.
- `initializeEditor` - Initialize an editor for test assertions.
- `openBlockSettings` - Opens the block settings of the current selected block.
- `changeAndSelectTextOfRichText` - Changes the text and selection of a RichText component.
- `changeTextOfRichText` - Changes the text of a RichText component.
- `pasteIntoRichText` - Paste content into a RichText component.
- `setupCoreBlocks` - Registers all core blocks or a specific list of blocks before running tests (once the tests are run, all registered blocks are unregistered).
- `setupMediaPicker` - Sets up Media Picker mock functions.
- `setupMediaUpload` - Sets up the media upload mock functions for testing.
- `changeTextOfTextInput` - Changes the text of a TextInput component.
- `triggerBlockListLayout` - Helper for ensuring that all items of a Block List component are rendered.
- `waitForModalVisible` - Waits for a modal to be visible.
- `waitForStoreResolvers` - Executes a function that triggers store resolvers and waits for them to be finished.
- `waitFor` - Custom implementation of the "waitFor" utility from `@testing-library/react-native` library.
- `withFakeTimers` - Set up fake timers for executing a function and restores them afterwards.
- `withReanimatedTimer` - Prepare timers for executing a function that uses the Reanimated APIs.

**NOTE:** The above helpers are also listed in a README file that is referenced in the [React Native Integration Test Guide](https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/react-native/integration-test-guide.md).

## Testing Instructions




1. Run command `npm run native test`.
2. Observe that all tests succeed.

## Screenshots or screencast 
N/A
